runtime/block_reduction.cu

```diff
@@ -215,96 +215,143 @@ __device__ void blockIterGroupedReduce(
   static_assert(
       N == 2 || N == 4 || N == 8 || N == 16,
       "N should be a valid vectorization factor, one of (2, 4, 8, 16)!");
+  static_assert(
+      !X_REDUCE && Y_REDUCE && !Z_REDUCE, "Only support reduction in Y dim!");
```

Comment:
nitpick here, if this is a static_assert and the logic below is only applied to this case, maybe we should put the assert in `csrc/codegen.cpp::genIterGroupedBlockReduction` and rename this runtime helper?

Reply:
changed accordingly.